include/nuttx/mm/mm.h

```diff
@@ -258,7 +259,8 @@ struct mm_heap_s
 
   struct mm_freenode_s mm_nodelist[MM_NNODES];
 
-#if defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__)
+#if defined(CONFIG_BUILD_FLAT) || defined(CONFIG_BUILD_PROTECTED) \
+  || defined(__KERNEL__)
```

Comment:
From your report, it's very important to keep mm_heap_s layout consistent in all the memory model. So I think the more safe fix is that:
1.Remove #ifdef/#endif around "struct mm_delaynode_s" and "struct mm_delaynode_s *mm_delaylist"
2.Rremove #ifdef/#endif around nuttx/mm/mm_heap/mm_initialize.c:181

Reply:
@xiaoxiang781216 Thanks for your comments. I will modify the code and push it later.

Reply:
@xiaoxiang781216 Should we remove #ifdef/#endif in mmheap/mm_malloc.c and mmheap/mm_free.c as well?

Reply:
Actually the initial code don't have #ifdef/#endif, but @liguiding found that up_interrupt_context/enter_critical_section/leave_critical_section doesn't expose to userspace. That is why you see this guard in the mainline. Since the userspace should never call kmm_free in the interrupt context, it's good to remove the delay free related code for userspace library until you report this issue.

Reply:
> Actually the initial code don't have #ifdef/#endif, but @liguiding found that up_interrupt_context/enter_critical_section/leave_critical_section doesn't expose to userspace. That is why you see this guard in the mainline. Since the userspace should never call kmm_free in the interrupt context, it's reasonable to remove the delay free related code from userspace library until you report this issue.

@xiaoxiang781216 Hmm, I understand. That's might be complicated than I though so I will not modify mm_malloc.c and mm_free.c this time.